### PR TITLE
Use proper `@Type` annotation for JSON type

### PR DESCRIPTION
**Pull request by hmlnarik**
Fixes: #16335

### CODE REVIEW

File: model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/group/entity/JpaGroupEntity.java [hunk 1]
```diff
@@ -62,7 +61,8 @@ public class JpaGroupEntity extends AbstractGroupEntity implements JpaRootVersio
     @Column
     private int version;
 
-    @Type(value = UserTypeLegacyBridge.class, parameters = @Parameter(name = UserTypeLegacyBridge.TYPE_NAME_PARAM_KEY, value = "jsonb"))
+//    @Type(value = UserTypeLegacyBridge.class, parameters = @Parameter(name = UserTypeLegacyBridge.TYPE_NAME_PARAM_KEY, value = "jsonb"))
+    @Type(value = JsonbType.class)
```
Comment (ahus1) [inline model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/group/entity/JpaGroupEntity.java:65]: Here's a commented out line that could be removed. 

The annotation can be written shorter without the `value=` as `@Type(JsonbType.class)` as the value is the default parameter.

**Review by ahus1 — CHANGES_REQUESTED** [review]
Hi @hmlnarik - it is great to see that the type mapping is again possible in a very clean way!

I reviewed the code and ran some model tests and manual tests, and it works great for me. 

In addition to the comment below, please clean the metadata fields that were part of the workaround in all the `*Metadata` classes, as they are no longer used.

```
private String metadataClass = this.getClass().getCanonicalName();
```

**Review by vramik — COMMENTED** [review]
Thank you @hmlnarik for the PR, it looks great. On top of what @ahus1 wrote I have just optional nitpick regarding import ordering. I've noticed that in some classes (e.g. `JsonbType`) there are import ordered alphabetically and the commit adds new imports to the bottom. On the other hand it's not true for all classes. It can stay as is of course, I just wanted to make a note about that. :-)

**Review by ahus1 — APPROVED** [review]
LGTM, thank you for this PR!

**Pull request merged**